src/schemaType/ArrayType.ts

```diff
@@ -34,12 +34,22 @@ class ArrayType extends BaseScalarArrayType {
 	}
 
 	/** Validate the array */
-	public async validate(value: unknown, document: Document): Promise<string[]> {
-		return (
-			await Promise.all(
-				ensureArray(value).map((arrayItem) => this.valueSchemaType.validate(arrayItem, document)),
-			)
-		).flat();
+	public async validate(value: unknown, document: Document): Promise<Map<string, string[]>> {
+		const errorsMap = new Map<string, string[]>();
+		await Promise.all(
+			ensureArray(value).map(async (arrayItem, index) => {
+				const result = await this.valueSchemaType.validate(arrayItem, document);
+
+				return result.forEach((message) => {
```

Comment:
I didn't catch this on the last pass, but I think this logic is more complicated that it needs to be. I think this could check to see if `result.length > 0` and set the map value to whatever is in `result` if it is.

Reply:
good idea. i will make this change.